src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -97,7 +98,17 @@ def get_annotation_args(annotation: Any, module: str, class_name: str) -> tuple[
     return getattr(annotation, "__args__", ())
 
 
-def format_annotation(annotation: Any, fully_qualified: bool = False, simplify_optional_unions: bool = True) -> str:
+def format_annotation(
+    annotation: Any,
+    fully_qualified: bool = False,
+    simplify_optional_unions: bool = True,
+    typehints_formatter: Callable[..., str] | None = None,
+) -> str:
+    # all options except for the formatter:
+    opts = {"fully_qualified": fully_qualified, "simplify_optional_unions": simplify_optional_unions}
+    formatted = None if typehints_formatter is None else typehints_formatter(annotation, **opts)
```

Comment:
Let's not build the opts dictionary but instead pass it in explicitly here as keyword arguments.

Reply:
OK, but to keep the lines short, I made it a partial in the `typing.Callable` branch